### PR TITLE
Gofmt code

### PR DESCRIPTION
**Pull request by fasmat**
- Formatted code with Go 1.18s gofmt
- Update dependencies that are not updated by dependabot

### CODE REVIEW

**Comment by fasmat** [comment]
`TestNip` keeps failing for some reason, generating less than 150 leaves when 150 are expected. I cannot reproduce this error locally. On my machine this test passes without problems. Any ideas why the test might fail on github?

File: hash/hash.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package hash
 
-import "github.com/spacemeshos/sha256-simd"
+import "github.com/minio/sha256-simd"
```
Comment (countvonzero) [inline hash/hash.go]: why this update?

File: shared/shared.go [hunk 2]
```diff
@@ -7,8 +7,8 @@ import (
 	"os"
 	"time"
 
+	"github.com/minio/sha256-simd"
 	"github.com/spacemeshos/merkle-tree"
-	"github.com/spacemeshos/sha256-simd"
```
Comment (countvonzero) [inline shared/shared.go]: ditto

File: hash/hash.go (hunk 1 above)
Comment (fasmat) [reply hash/hash.go]: This seems to be the only repo where we are still using our (outdated) fork over the upstream version if the library.

Comment (countvonzero) [reply hash/hash.go]: was the test failure caused by minio?

Comment (fasmat) [reply hash/hash.go]: Unfortunately not, it seems like golang.org/x/net is the culprit :-( at least if I update the dependency tests start failing. I will still put the migration from `github.com/spacemeshos/sha256-simd` to `github.com/minio/sha256-simd` into its own PR.

**Review by countvonzero — APPROVED** [review]

**Pull request merged**